src/libcore/ops/range.rs

```diff
@@ -696,6 +696,28 @@ pub enum Bound<T> {
     Unbounded,
 }
 
+impl<T: Clone> Bound<&T> {
+    /// Map a `Bound<&T>` to a `Bound<T>` by cloning the contents of the bound.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::ops::Bound::*;
+    /// use std::ops::RangeBounds;
+    ///
+    /// assert_eq!((1..12).start_bound(), Included(&1));
+    /// assert_eq!((1..12).start_bound().cloned(), Included(1));
+    /// ```
+    #[unstable(feature = "bound_cloned", issue = 61356)]
+    fn cloned(&self) -> Bound<T> {
+        match self {
```

Comment:
```suggestion
        match *self {
```

A dereference here prevents the implicit `&&T` in the pattern.